CIP-0037/CIP-0037.md

```diff
@@ -0,0 +1,338 @@
+as follows: 
+* Check that n <= t - k 
+* n (log_2(t) hash computations + equality checks) (this is a very pessimistic upper bound. In practice it's much less)
+* n G_1 additions and 1 G_1 negation
+* 2 pairing evaluations + equality check
+
+## Mithril (using trivial proving system)
+Let k be the number of submitted signatures (as above, k is a security parameter determining the number of 
+required signatures). However, in mithril k << N where N is the total number of eligible signers. Mithril 
+verification goes as foollows: 
+* Check the k is large enough
+* k G_1 additions
+* k G_2 additions
+* k (log_2(N) hash computations + equality checks)
+* 2 pairing evaluations + equality check
+
+There are some additional t checks of lottery winning (similar to VRF). These are non crytpgraphic operations 
+(rather inequality checks of indices, and checks that some hash computation is below a threshold).
+
+## Coconut certs
+We use this as an example for anonymous credentials. There is no decision on what type of construction we will 
+eventually use. Coconut credentials (and other types of credentials we are looking at) are built in a way that 
+it is efficient to prove statements about attributes. To this end, it is required to build, and be able to verify, 
+relations over discrete logarithm values. However, for sake of simplicity in this computation, we consider the 
+simplest form of anonymous credentials, which contains no attributes. We not that proving statements regarding 
+attributes does not require further pairing evaluations. 
+* Proof verification to a relation involving elements in G_1 and G_2. This verification does not require the 
+* pairing evaluation, but does require G_1 and G_2 additions and multiplications. 
+* 2 pairing evaluations + equality check
+
+The complexity of the statement one can prove with anonymous credentials will be bounded by plutus computational 
+limits. However, it is hard to make individual functions for each possible statement accepted. 
+
+## Plonk verification
+Plonk verification is quite complex, and might differ depending on the number of custom gates, or specific wire 
+connections. In this section we expose the verifier complexity as described 
+[here](https://hackmd.io/Vr0DUYPCR0iWADpO8kpkAA?view#Verifier-Algorithm). We count every challenge computation 
+as a single hash evaluation. We omit the `scalar X scalar` multiplications and additions. As we can see above, 
+these operations are insignificant with respect to group operations. We assume that point validation is performed 
+by the external C library, and therefore do not count that in this estimate. The computation is the following: 
+
+* 6 hash computations
+* 1 vanishing polynomial (1 scalar exponantiation)
+* Quotient polynomial evaluation (1 scalar inverse)
+* First part of batch poly commitment (7 scalar mults and 6 additions in G_1)
+* Fully batched poly commitment (7 scalar mults and 8 additions over G_1)
```

Comment:
I've counted 9 scalar mults in the previous in the previous version.
Now: 6 scalar mults and 5 additions in G_1

Reply:
I count 9 and 9 in the first part of the poly com, and 5 and 5 in the second part. Thanks for the suggestion of referencing the paper, will do that 👍